docker-compose.yml

```diff
@@ -23,13 +27,18 @@ services:
             - XMPP_BOSH_URL_BASE=http://xmpp.meet.jitsi:5280
             - XMPP_GUEST_DOMAIN
             - XMPP_MUC_DOMAIN
+            - XMPP_RECORDER_DOMAIN
             - TZ
         networks:
             meet.jitsi:
+                aliases:
```

Comment:
Why is this needed?

Reply:
for the the same reason as before. it maps domain name of web container "meet.jitsi" to the internal "dns"

Reply:
But it isn't, if we focus on case 1, right? Let's start there.

Reply:
It's needed for case 1

Reply:
This alias get's confusing, can you please change it to "web.meet.jitsi"?

Reply:
Yes, but changing the name will affects:
```
"xmpp_environments": [
{
...
 "xmpp_domain": "{{ .Env.XMPP_DOMAIN }}" 
...
}
``` 
in jibri/rootfs/defaults/config.json
So, I guess, by logic it must be xmpp_domain and we already have it as "meet.jitsi" isn't it?

Reply:
Maybe @bbaldino can enlighten us here. What should that point to?

Reply:
Given that we set `xmpp_server_hosts` to the actual XMPP server, I wonder what's going on. @netaskd what happens if you remove the alias?

Reply:
jibri container won't find https://meet.jitsi/

Reply:
If that is the case, then that config option is really ill named. Anyway, plase alias the container to web.meet.jitsi and let's pass an extra env variable: XMPP_WEB_SERVER.

Reply:
When I change it I get the error
```
jibri_1_1838a0eed8da | 2019-01-31 18:09:10.093 SEVERE: [40] org.jitsi.jibri.api.xmpp.XmppApi.run() Error starting Jibri service : java.lang.StringIndexOutOfBoundsException: String index out of range: -1 with stack: 
jibri_1_1838a0eed8da | java.lang.String.substring(String.java:1967)
jibri_1_1838a0eed8da | java.lang.String.subSequence(String.java:2003)
jibri_1_1838a0eed8da | org.jitsi.jibri.util.XmppUtilsKt.getCallUrlInfoFromJid(XmppUtils.kt:38)
jibri_1_1838a0eed8da | org.jitsi.jibri.api.xmpp.XmppApi.handleStartService(XmppApi.kt:276)
jibri_1_1838a0eed8da | org.jitsi.jibri.api.xmpp.XmppApi.access$handleStartService(XmppApi.kt:71)
jibri_1_1838a0eed8da | org.jitsi.jibri.api.xmpp.XmppApi$handleStartJibriIq$1.run(XmppApi.kt:200)
jibri_1_1838a0eed8da | java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
jibri_1_1838a0eed8da | java.util.concurrent.FutureTask.run(FutureTask.java:266)
jibri_1_1838a0eed8da | java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
jibri_1_1838a0eed8da | java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
jibri_1_1838a0eed8da | java.lang.Thread.run(Thread.java:748)
```
I guess it because some function try to handle config option: 
```
"room_jid_domain_string_to_strip_from_start"
```

Reply:
Hum. Let's see if @bbaldino can enlighten us. I see xmppDomain is used here: https://github.com/jitsi/jibri/blob/0e96134dcc5370d4668533e160d573f13436f591/src/main/kotlin/org/jitsi/jibri/util/XmppUtils.kt#L42 But that seems a bit off to me, since the web URL may be completely diffrent to the XMPP domain, as we are seeing here.

Reply:
web url is currently defined by the xmpp domain, because we derive it from looking at the domain of the jid in the jibri request.  i don't think jicofo actually knows the web url of the conference.  @aaronkvanmeerten probably has some good info here.

Reply:
the error from here:
https://github.com/jitsi/jibri/blob/0e96134dcc5370d4668533e160d573f13436f591/src/main/kotlin/org/jitsi/jibri/util/XmppUtils.kt#L38
Because it don't expected domain like web.meet.jitsi only like meet.jitsi
muc.meet.jitsi (from jid) can't be parsed to subdomain from web.meet.jitsi, only like muc.web.meet.jitsi can be.

Reply:
That's unfortunate, because they are not really related :-/

@netaskd please leave the alias there then. Thanks for trying things out!